app/models/metric/targets.rb

```diff
@@ -58,16 +56,46 @@ def self.capture_container_targets(zone, _options)
     targets
   end
 
-  def self.capture_vm_targets(targets, parent_class, options)
-    return Vm.none if options[:exclude_vms]
-    enabled_parents = targets.select do |t|
-      t.kind_of?(parent_class) &&
-        t.kind_of?(Metric::CiMixin) &&
-        t.perf_capture_enabled? &&
-        t.respond_to?(:vms)
+  def self.preload_infra_targets_data(zone, options)
+    # Preload all of the objects we are going to be inspecting.
+    includes = {:ext_management_systems => {:hosts => {:ems_cluster => :tags, :tags => {}}}}
+    includes[:ext_management_systems][:hosts][:storages] = :tags unless options[:exclude_storages]
+    includes[:ext_management_systems][:hosts][:vms] = {} unless options[:exclude_vms]
```

Comment:
If you wanted to extract a method "includes_for_preload" here, you might fix those complexity warnings